src/Views/FirmwareView.vala

```diff
@@ -76,14 +73,21 @@ public class About.FirmwareView : Gtk.Stack {
         add (grid);
         add (progress_view);
 
-        fwupd.on_device_added.connect (on_device_added);
-        fwupd.on_device_error.connect (on_device_error);
-        fwupd.on_device_removed.connect (on_device_removed);
+        try {
+            var fwupd_client = new Fwupd.Client ();
+            FirmwareClient.connect.begin (fwupd_client, (obj, res) => {
+                fwupd_client.device_added.connect (on_device_added);
```

Comment:
The `try/catch` has to surround the `async_method.end` call which is missing here.